regolith/tools.py

```diff
@@ -1282,9 +1245,82 @@ def fragment_retrieval(coll, fields, fragment, case_sensitive = False):
                     break
     return ret_list
 
+
 def get_id_from_name(coll, name):
     person = fuzzy_retrieval(coll,["name", "aka", "_id"], name,
                              case_sensitive=False)
     if person:
         return person["_id"]
-    else: return None
\ No newline at end of file
+    else: return None
+
+
+def is_fully_appointed(person, begin_date=None, end_date=None):
+    """Checks if a collection of appointments for a person is valid and fully loaded
+        for a given interval of time
+
+        Parameters
+        ----------
+        person: dict
+            The person whose appointments need to be checked
+        begin_date: datetime, string, optional
+            The start date of the interval of time to check appointments for
+        end_date: datetime, string, optional
+            The end date of the interval of time to check appointments for
+
+        Returns
+        -------
+        bool:
+            True if the person is fully appointed and False if not
+
+        Examples
+        --------
+        >>> appts = [{"begin_year": 2017, "begin_month": 6, "begin_day": 1, "end_year": 2017, "end_month": 6,\
+         "end_day": 15, "grant": "grant1", "loading": 1.0, "type": "pd", }, {"begin_year": 2017, "begin_month": 6, \
+         "begin_day": 17,  "end_year": 2017,  "end_month": 6, "end_day": 30, "grant": "grant2", "loading": 1.0, \
+         "type": "pd",} ]
+        >>> aejaz = {"name": "Adiba Ejaz", "_id": "aejaz", "appointments": appts}
+        >>> is_fully_appointed(aejaz, "2017-06-01", "2017-06-30")
+
+        In this case, we have an insufficient loading and minimum of 0.0 on 2017-06-16, hence it would return False and
+        print "aejaz is short of full loading with minimim 0.0 on 2017-06-16".
+        """
+
+    status = True
+    appts = person.get('appointments')
+    earliest = get_dates(appts[0])['begin_date']
+    latest = get_dates(appts[0])['end_date']
+    for appt in appts:
+        dates = get_dates(appt)
+        if dates['begin_date'] < earliest:
+            earliest = dates['begin_date']
+        if dates['end_date'] > latest:
+            latest = dates['end_date']
+    begin_date = earliest if not begin_date else begin_date
+    end_date = latest if not end_date else end_date
+    if begin_date > end_date:
+        raise ValueError("invalid begin and end dates")
+    if isinstance(begin_date, str):
+        begin_date = date_parser.parse(begin_date).date()
+    if isinstance(end_date, str):
+        end_date = date_parser.parse(end_date).date()
+    timespan = end_date - begin_date
+    loading, max, min, help = [], 0.0, 1.0, ''
+    for x in range(0, timespan.days):
+        loading.append(0.0)
+        day = begin_date + relativedelta(days=x)
+        for appt in appts:
+            if is_current(appt, now=day):
+                loading[x] = loading[x] + appt.get("loading")
```

Comment:
couldn't you everywhere in here just have loading as a scalar variable, i.e., ``day_loading = day_loading + appt.get("loading")`` then at the very end have ``loadings.append(loading)``?  

What behavior do we want for the function? Do we want it to report the first bad loading it finds?  In that case, as soon as we find a bad one we can print the ``"WARNING: {} exceeds full loading on {}".format(person.get('_id'), max, str(day))"`` or whatever.  Do we want it to find the one, worst, bad loading (which I think is what your version is doing). Or do we want it to print all the bad loadings?

From my perspective, the first is fine but the easiest to debug my appointments might be to have a list of all the bad appointments.  This could make a huge long list if there are many bad days, so more elegant would be logic that would find a day that is not fully appointed and print just once for that whole "appointment gap", but in the spirit of agile programming, just dumping a long list of bad days in print statements is quick and easy.    None of these UCs require you to build the loadings array and find its max I think, so the code in here could be much less (which means easier to understand and less to maintain)

Reply:
Done for the loading array - it seems I didn't need it at all! I have also included functionality to print "appointment gaps" and tested this locally by returning the `messages` in a list and checking in my tests, but I have pushed a version which instead prints to stdout and returns  a bool! the different edge cases make the code a little more obtuse, but the helper messages are condensed and quite useful. let me know what you prefer!

Reply:
I like your solution!   It will be easier for me to test when I try it.   As long as you are confident that it is working correctly I am good.